index.js

```diff
@@ -141,44 +191,60 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+let num = 0
+function takeATicketNumber(line){
+  num++
+  line.push(num)
+  return `Welcome. You are ticket number ${num}`
+}
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 1.`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 2.`
+console.log("%c----------", "color: red") 
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
+
+
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+const newLine = []
 
-// const newLine = []
-// const takeATicketNumberFunc = ticketNumberGeneratorFunc()
-// console.log(takeATicketNumberFunc(newLine))
+function ticketNumberGeneratorFunc(){
+  number = 0
+  function takeATicketNumberFunc(array){
+    number++
+    array.push(number)
+    return `Welcome. You are ticket number ${number}`
+  }
+  return takeATicketNumberFunc
+}
+
+const takeATicketNumberFunc = ticketNumberGeneratorFunc()
```

Comment:
Nice job! Only thing is don't forget to declare your number variable with `let`. If you don't use a keyword, the variable will become global